cloud formation/template.yaml

```diff
@@ -0,0 +1,149 @@
+
+Parameters:
+  ApiStagename:
+    Type: String
+    Default: develop
+
+Resources:
+  S3Bucket:
+    DeletionPolicy: Retain
+    Type: 'AWS::S3::Bucket'
+    Description: Creating Amazon S3 bucket from CloudFormation
+    Properties:
+      BucketName: bucket789987
+      AccessControl: PublicReadWrite
+  SampleBucketPolicy:
+    Type: AWS::S3::BucketPolicy
+    Properties:
+      Bucket: !Ref S3Bucket
+      PolicyDocument:
+        Version: 2012-10-17
+        Statement:
+            - Action:
+                - 's3:PutObject'
+              Effect: Allow
+              Resource: !Join
+                - ''
+                - - 'arn:aws:s3:::'
+                  - !Ref S3Bucket
+                  - /*
+              Principal: '*'
+              Condition:
+                StringLike:
+                  'aws:Referer':
+                    - 'http://www.example.com/*'
+                    - 'http://example.net/*'
+
+  permission:
+    Type: AWS::Lambda::Permission
+    Properties:
+      FunctionName: !Ref writeLogsToS3
```

Comment:
You're creating the lambda function down how can you reference it here? this block should go down after you've created the lambda